test_search_count_and_cursor failure with invalid objects count after search

## Current Behavior
Test failure at "Search object" step.
```
RuntimeError: Command: ./neofs-cli --config /home/runner/work/neofs-testcases/neofs-testcases/neofs-testcases/wallet_config.yml object searchv2 --rpc-endpoint 'localhost:50463' --wallet '/home/runner/work/neofs-testcases/neofs-testcases/neofs-testcases/test-run-2025-06-30-16-49-55-400228/wallet-d05adfc2-56a9-43ac-b5e0-4e8e94c92b49.json' --cid 'DksFiJe9sxLb4Nd6UpyCFhcpjxdLLEGxPpKYcBVxpbv4' --filters '$Object:creationEpoch GE 0' --attributes '$Object:creationEpoch' --count 2 --cursor '0123_???!##'
Error:
return code: 1
output: rpc error: status: code = 1024 message = invalid cursor: decode Base64: illegal base64 data at input byte 4
```

## Expected Behavior
Green.

## Possible Solution
The cursor is obviously wrong, how did we end up with this cursor there is a question.

## Steps to Reproduce
https://rest.fs.neo.org/HXSaMJXk2g8C14ht8HSi7BBaiYZ1HeWh2xnWPGQCg4H6/1199-1751307611/index.html#suites/0e3180b9561587482b2f1488d7674deb/2251ad3ffdc0164f/

## Regression
Doesn't look like.

## Your Environment
* Version of the product used: master testcases, 0.47.1 node

@roman-khimov the problem here is in this request (the one you provided with an invalid cursor is expected, and is not a cause of the failure):
```
COMMAND: ./neofs-cli --config /Users/runner/work/neofs-node/neofs-node/neofs-testcases/wallet_config.yml object searchv2 --rpc-endpoint 'localhost:59932' --wallet '/Users/runner/work/neofs-node/neofs-node/neofs-testcases/test-run-2025-07-01-09-54-00-982498/wallet-023d4e7f-2ff2-499a-99f2-7544ea220f37.json' --cid '4dLQLM8bVRn112vnBFwwQms7e7mtaDXr4XGJAraDtkGX' --filters '$Object:creationEpoch GE 0' --attributes '$Object:creationEpoch,$Object:objectType' --cursor 'JE9iamVjdDpjcmVhdGlvbkVwb2NoAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABLY8SaMvZHiXBg0fZIQtnJqhPM5K+OljiekHe3CBWdE5'
RETCODE: 0

STDOUT:
Found 0 objects.

STDERR:

Start / End / Elapsed	 10:31:11.979771 / 10:31:12.243033 / 0:00:00.263262
```
that is performed after multiple object removals and waiting for a new epoch:
<img width="839" alt="Image" src="https://github.com/user-attachments/assets/4efccdbc-a0c2-45f3-82b6-2edfa8db0d0c" />

Test expects to see tombstone objects:
```
        tombstone_objects, _ = search_objectv2(
            rpc_endpoint=neofs_env.sn_rpc,
            wallet=default_wallet.path,
            cid=cid,
            shell=neofs_env.shell,
            filters=["$Object:creationEpoch GE 0"],
            attributes=["$Object:creationEpoch", "$Object:objectType"],
            cursor=first_cursor,
        )
&gt;       assert len(tombstone_objects) &gt; 0, "invalid objects count after search"
E       AssertionError: invalid objects count after search
E       assert 0 &gt; 0
E        +  where 0 = len([])
```
But for some reason, sometimes they are not there. 
Do you have any ideas how can it be fixed from the test perspective?

--- a/pytest_tests/tests/services/rest_gate/test_rest_searchv2.py
+++ b/pytest_tests/tests/services/rest_gate/test_rest_searchv2.py
@@ -893,23 +893,33 @@ def test_search_count_and_cursor(
             neofs_env.sn_rpc,
         )
 
-    neofs_epoch.tick_epoch_and_wait(neofs_env)
-    neofs_epoch.tick_epoch_and_wait(neofs_env)
+
+def test_search_tombstone_objects(default_wallet: NodeWallet, container: str, neofs_env: NeoFSEnv):
+    cid = container
+    file_path = generate_file(neofs_env.get_object_size("simple_object_size"))
+
+    oid = put_object_to_random_node_via_rest_gw(
+        cid=cid,
+        path=file_path,
+        endpoint=f"http://{neofs_env.rest_gw.endpoint}/v1",
+    )
+
+    found_objects, _ = search_objectv2(
+        cid=cid,
+        filters=["$Object:creationEpoch GE 0"],
+        attributes=["$Object:creationEpoch"],
+        neofs_env=neofs_env,
+    )
+
+    assert len(found_objects) == 1, "invalid objects count after search"
+
+    delete_object(default_wallet.path, cid, oid, neofs_env.shell, neofs_env.sn_rpc)
 
     tombstone_objects, _ = search_objectv2(
         cid=cid,
-        filters=["$Object:creationEpoch GE 0"],
+        filters=["$Object:objectType EQ TOMBSTONE"],
         attributes=["$Object:creationEpoch", "$Object:objectType"],
-        cursor=first_cursor,
         neofs_env=neofs_env,
     )
-    assert len(tombstone_objects) > 0, "invalid objects count after search"
-
-    with pytest.raises(Exception, match=".*wrong primary attribute.*"):
-        not_tombstone_objects, _ = search_objectv2(
-            cid=cid,
-            filters=["$Object:objectType NE TOMBSTONE"],
-            attributes=["$Object:objectType"],
-            cursor=first_cursor,
-            neofs_env=neofs_env,
-        )
+    assert len(tombstone_objects) == 1, "invalid tombstone objects count after search"
+    assert tombstone_objects[0]["id"] != oid, "invalid tombstone object returned from search"

